tests/test.rs

```diff
@@ -596,4 +596,27 @@ export type Test=(types.Test3&{"a":string;"b":types.ExternalStringWrapper;"c":st
 "#
         );
     }
+
+    #[test]
+    fn emit_expr() {
+        #[derive(Serialize, TypeDef)]
+        struct Message<T> {
+            op: String,
+            payload: T,
+        }
+
+        type StringMessage = Message<String>;
+        type BytesMessage = Message<Vec<u8>>;
+
+        let options = DefinitionFileOptions::default();
+        let mut s = vec![];
+        StringMessage::INFO.emit_expr(&mut s, options).unwrap();
+        assert_eq!("types.Message<string>", std::str::from_utf8(&s).unwrap());
+        let mut s = vec![];
+        BytesMessage::INFO.emit_expr(&mut s, options).unwrap();
+        assert_eq!(
+            "types.Message<(types.U8)[]>",
```

Comment:
Hmm, this example highlights a potential problem with this API: if your type has dependencies on other types, where is it getting those definitions from? How is this generated code useful without having a definition for `types.U8` somewhere accessible?

You said you're using this to create TypeScript method stubs for an autogenerated RPC client. Can you explain more about how that works? Maybe there's a different way to accomplish what you want.